src/sandstorm/sandstorm-http-bridge.c++

```diff
@@ -1274,6 +1338,27 @@ private:
     lines.add(kj::str(""));
   }
 
+  static bool checkIsHttpToken(kj::StringPtr val) {
```

Comment:
You have these helpers returning bool but I think it may make sense for them to throw exceptions. I can't think of a case where it's important to gracefully recover from garbage header names or values.

In fact I'd have the helpers return their parameter if it's valid and throw if not. That way we can easily add calls to these helpers in lots of other places above where we ought to have had them all along, to sanitize other input strings.

OTOH, if a header name is merely missing from the whitelist, then we should drop it rather than throw, otherwise it could be a compatibility issues.